Quick_Deploy/HuggingFaceTransformers/create_repository.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import os
+import argparse
+import json
+import shutil
+import subprocess
+from transformers import pipeline, AutoTokenizer
+
+
+class ServerBuilder:
```

Comment:
I'm not sure how I feel about this builder script. Part of me would rather focus on making the huggingface a backend or just simply intuitive to use.

For example, the config files should have sensible default values for model name and max length so users can run it as-is, but also see intuitive places to replace the values with their own values. But if you do that, then this script gets harder to write because it's harder to find/replace the "placeholder" tags you put.

Reply:
If this script were to stay, I'd rather see the default values in `config.pbtxt` be actually usable values, and make the folders here in the repo actually in correct model repository structure by default.

If you want to generate a config file with a CLI like this, you could just overwrite the config file from a template rather than trying to find/replace.

Reply:
At a minimum, I would lean towards separating this PR into 2. I think we should add the huggingface example model with sensible default values in config and files in model_repository structure that can just be run as-is.

Separately, can discuss a "builder" script or other deployment tool along those lines. I wouldn't bundle them here for now.

Reply:
Agree, directionally the tutorial should be easy to follow / customize / few steps - but shouldn't hide too much w.r.t to automation - I think that will come when we add support into either additional tools or backends.